lib/Doctrine/Common/Cache/CacheProvider.php

```diff
@@ -126,10 +138,20 @@ public function contains($id)
 
     /**
      * {@inheritdoc}
+     *
+     * @throws \Doctrine\Common\Cache\Exception\LifeTimeException
      */
     public function save($id, $data, $lifeTime = 0)
     {
-        return $this->doSave($this->getNamespacedId($id), $data, max($lifeTime, 0));
+        if (! is_int($lifeTime)) {
```

Comment:
numeric strings should IMO not fail. otherwise the BC break might be huge for some people.